CMakeLists.txt

```diff
@@ -189,6 +189,8 @@ if(ENABLE_ZLIB_SUPPORT)
     endif(ZLIB_FOUND)
 endif(ENABLE_ZLIB_SUPPORT)
 
+include_directories(library)
```

Comment:
In the commit message you mention to add "library/" as a known location of header files to build and test scripts but it seems to me that this is adding "library/" as an header file location not just to build and test scripts but probably for any compilation. I don't see a similar change in the root Makefile. Maybe this should rather be in tests/CMakeLists.txt?

Reply:
This would be `library/CMakeLists.txt` as well as `tests/CMakeLists.txt`.

Reply:
I did it this way because mbed-crypto has this line. If that's wrong, we can change it here and remember to process it correctly during the merge, or remember it for after the merge.

The practical difference is whether `programs` has access to internal headers. Sample programs shouldn't, but test programs might need to. We should separate those but that's way out of scope.